### ISSUE TITLE
GitHub workflow to add PR maintainer checklist

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#183]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue would complete the work already done in [.github/workflows/pr_checklist.yaml](https://github.com/activist-org/activist/blob/main/.github/workflows/pr_checklist.yaml) to set up a PR checklist for activist. As of now this would just add a check for maintainers to make sure that the contributors account is set up to be added to the contributors graph. From there we could add further points like updating the changelog when activist shifts to versioned releases :)

### Contribution

Happy to discuss and support someone on this!

**Comment by andrewtavis** [comment]
Looks like the action [actions-comment-pull-request](https://github.com/thollander/actions-comment-pull-request) could be used for this, with [this blog post](https://jacobtomlinson.dev/posts/2022/commenting-on-pull-requests-with-github-actions/) being a good source for getting it up and running :)

**Comment by andrewtavis** [comment]
[This PR](https://github.com/activist-org/activist/pull/186) shows the flow of the PR checklist. The message is displayed only after the PR is opened as desired :) Final edits are needed for the texts and then we’ll be good to close.

**Comment by andrewtavis** [comment]
Closed in 9d2ba97 :)

**Comment by andrewtavis** [comment]
This workflow failed on #187, so I'm removing it for now :) Issue is reopened and we can look into it later.

**Comment by andrewtavis** [comment]
It appears that remote repos don't have write permissions regardless of them having is directly assigned. The `.yaml` file has been converted to [workflows/pr_maintainer_checklist.txt](https://github.com/activist-org/activist/blob/main/.github/workflows/pr_maintainer_checklist.txt) for now until we can get it up and running, as just commenting it out was causing it to still run nothing.

**Comment by andrewtavis** [comment]
It could be something in the repos settings is breaking all this 🤔

**Comment by linusha** [comment]
@andrewtavis have tried seen this?

> You can use the permissions key to add and remove read permissions for forked repositories, but typically you can't grant write access. The exception to this behavior is where an admin user has selected the Send write tokens to workflows from pull requests option in the GitHub Actions settings. For more information, see "[Managing GitHub Actions settings for a repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-private-repository-forks)." (from [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions))

From a quick glance it looks to me as if enabling that (Send write tokens to workflows from pull requests) option might solve the issue?

**Comment by andrewtavis** [comment]
I was expecting it was something like this :) Thanks @linusha! Nice to chat with you on here 😊😊 Will try it out soon and report back once I give a test. Would you be willing to do a trial PR for it to see if it works?

**Comment by linusha** [comment]
> I was expecting it was something like this :) Thanks @linusha! Nice to chat with you on here blushblush Will try it out soon and report back once I give a test. Would you be willing to do a trial PR for it to see if it works?

Yes, absolutely. Just give me a :rocket: reaction here, once you had time to change the setting, ok? :slightly_smiling_face:

**Comment by andrewtavis** [comment]
Hey @linusha 👋 I'm checking the options and it seems that write permissions should be there already given the settings. The only thing I'm seeing is that forks might need approval to run for first time contributors 🤔

Looking at the note within [controlling-changes-from-forks-to-workflows-in-public-repositories](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#controlling-changes-from-forks-to-workflows-in-public-repositories) it says the following:

> Note: Workflows triggered by `pull_request_target` events are run in the context of the base branch. Since the base branch is considered trusted, workflows triggered by these events will always run, regardless of approval settings. For more information about the `pull_request_target` event, see "[Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)."

The current setup is on `pull_request` rather than `pull_request_target`, so this might make sense to change? A similar suggestion is made [in this GitHub security blog post](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/#:~:text=The%20main%20differences%20between%20the,but%20not%20from%20external%20forks.) :) Making the change to `pull_request_target` now and changing the file back to `.yaml`. Would be great if you could check it at your convenience 😊

**Comment by linusha** [comment]
@andrewtavis I think it works?! :slightly_smiling_face: 
https://github.com/activist-org/activist/pull/189 has the comment :tada: 

When you confirmed that everything works as expected, please feel more than free to just close the PR :joy:

**Comment by andrewtavis** [comment]
Thanks, @linusha! Added you as an assignee to give some credit for the discussion and the PR 😊 Officially closed via d02c26c 🤖🚀

**Comment by andrewtavis** [comment]
Thanks for the help!!
